### PR TITLE
US4598: Deploy vault TLS assets to /run/edgex/secrets/edgex-vault

### PR DESCRIPTION
**Pull request by jim-wang-yutsung**
Implementation of deploy vault TLS things with sentinel files

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

### CODE REVIEW

**Review by bnevis-i — APPROVED** [review]
LGTM

File: internal/pkg/pkiinit/option/optionHelpers.go [hunk 1]
```diff
@@ -164,3 +170,47 @@ func zeroOutFile(fileToErase string) error {
...
+func markComplete(dirPath string) error {
+	// recursively walk through all sub-directories until reach the leaf node
+	// to write the sentinel files
+	fileDescs, readErr := ioutil.ReadDir(dirPath)
+	if readErr != nil {
+		return readErr
+	}
+
+	for _, fileDesc := range fileDescs {
+		aFilePath := filepath.Join(dirPath, fileDesc.Name())
```
Comment (eno-intel) [inline internal/pkg/pkiinit/option/optionHelpers.go:197]: not the biggest deal but is there a better variable name for this other than aFilePath.

File: internal/pkg/pkiinit/option/optionHelpers.go [hunk 2]
```diff
@@ -131,7 +134,10 @@ func getPkiCacheDirEnv() string {
 }
 
 func deploy(srcDir, destDir string) error {
-	return copyDir(srcDir, destDir)
+	if err := copyDir(srcDir, destDir); err != nil {
+		return err
+	}
+	return markComplete(destDir)
```
Comment (eno-intel) [inline internal/pkg/pkiinit/option/optionHelpers.go:140]: deploy(...) is also called during other options, like import and cache. do we want to create a sentinel file in those cases as well?

Comment (jim-wang-yutsung) [reply internal/pkg/pkiinit/option/optionHelpers.go:140]: yes

File: internal/pkg/pkiinit/option/optionHelpers.go (hunk 1 above)
Comment (jim-wang-yutsung) [reply internal/pkg/pkiinit/option/optionHelpers.go:197]: not sure what other better variable name I can think of... `aFilePath` pretty much fits its purpose here

**Review by eno-intel — APPROVED** [review]

**Review by bnevis-i — APPROVED** [review]
LGTM

**Pull request merged**